src/material-experimental/mdc-slider/slider-thumb-decorator.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  Output,
+  ViewChild,
+  ViewEncapsulation,
+} from '@angular/core';
+import {Thumb} from '@material/slider';
+
+/**
+ * Handles displaying the slider knobs and their value indicators.
+ */
+@Component({
+  selector: 'mat-slider-start-thumb-decorator, mat-slider-end-thumb-decorator',
+  templateUrl: 'slider-thumb-decorator.html',
+  host: {
+    'class': 'mdc-slider__thumb',
+    '(mouseenter)': '_mouseenter.emit()',
+    '(mouseleave)': '_mouseleave.emit()',
+  },
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MatSliderThumbDecorator {
+  /** Whether the slider is discrete. */
+  @Input() isDiscrete: boolean;
+
+  /** The text content of the value indicator for a discrete slider. */
+  @Input()
+  get valueIndicatorText(): string { return this._valueIndicatorText; }
+  set valueIndicatorText(v: string) {
+    this._valueIndicatorText = v;
+    this._cdr.detectChanges();
```

Comment:
Why is this call necessary? Angular should do it by default when an `@Input()` changes, even though it's an `OnPush` component.

Reply:
I'm not sure why it's necessary but I am sure that it's necessary. Without this call, the value indicator text doesn't update properly _while_ dragging - only after the drag ends. Changing the value using arrow keys will not work without this call, either.

Again, I have no idea why this is the case. Let me know if you'd like to take a deeper look and we can schedule a 1:1 to figure out why this happens.

Reply:
+1, it's an anti-pattern to call `detectChanges` like this because it detects changes _immediately_ instead of coallessing multiple changes into a single change detection round